app.py

```diff
@@ -734,8 +734,33 @@ def jurisdiction_retrieval_list(election_id, jurisdiction_id, round_num):
                     .order_by(AuditBoard.name, Batch.name, SampledBallotDraw.ballot_position, SampledBallotDraw.ticket_number) \
                     .all()
 
+    ballot_to_sampled = {}
     for ballot_draw, batch, audit_board in ballots:
-        retrieval_list_writer.writerow([batch.name, ballot_draw.ballot_position, batch.storage_location, batch.tabulator, ballot_draw.ticket_number, audit_board.name])
+
+        index = (batch.name, ballot_draw.ballot_position)
+        if index in ballot_to_sampled:
+            ballot_to_sampled[index]['ticket_numbers'].append(ballot_draw.ticket_number)
+        else:
+            ballot_to_sampled[index] = {
```

Comment:
too much code here, why stash these in a dictionary that we then immediately pull out? How about just storing the array of `ballot_draw` objects and then pulling them out with attributes, not keyed dictionary entires? Would shrink this code by 10-15 lines.

Reply:
I did it for logical simplicity (i.e. laziness :sweat_smile:). I'll take a look at refactoring.